application/models/services/PermissionManager.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Management of Permission
+ * @version 0.1.0
+ * @author Denis Chenu
+ * @copyright LimeSurvey
+ */
+
+namespace LimeSurvey\Models\Services;
+
+use LSHttpRequest;
+use LSWebUser;
+use Permission;
+
+/**
+ * Service class for managing permission
+ */
+class PermissionManager
+{
+
+    /** @var LSHttpRequest */
+    private $request;
+
+    /** @var LSWebUser */
+    private $user;
+
+    /**
+     * @param LSHttpRequest $request
+     * @param LSWebUser $user
+     */
+    public function __construct(
+        LSHttpRequest $request,
+        LSWebUser $user
+    ) {
+        $this->request = $request;
+        $this->user = $user;
+    }
+
+    /**
+     * get the permission data
+     * @param string $modelName the model name
+     * @param interger $modelId the model name
+     * @param integer $userId for this user id
+     * @return array[] 
+     */
+    public function getPermissionData($modelName, $modelId, $userId = null)
+    {
+        $aObjectPermissions = Permission::model()->getEntityBasePermissions($modelName);
```

Comment:
If you want to be able to test this with a database scaffold, you need to either inject the Permission model in the constructor, OR factor out access to permission into separate methods and then mock those functions in the test.

Reply:
_without_ database scaffold, sorry for the typo

Reply:
Permission::model()->getEntityBasePermissions didn't use database but model definition. 

See https://github.com/LimeSurvey/LimeSurvey/blob/186bfd8870083724a227d569c13a1c395b01f804/application/models/LSActiveRecord.php#L464 i can use it directly : OK ?

Reply:
But still have `Permission::model()->hasPermission($modelId, $modelName, $sPermission, $crud, $this->user->id),` …

Reply:
https://github.com/LimeSurvey/LimeSurvey/pull/1709/commits/c56bb3772602e53b43a931e663f8949da8ded2aa  ?